### ISSUE TITLE
Aircrack-ng master doesn't compile

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1654]
**Reported by anonymous on 11 Dec 2015 22:18 UTC**

On Ubuntu 15.10 x64, with latest git master (because I don't know how to svn):

```
jvoisin@kaa 23:15 /opt/aircrack-ng git pull
Already up-to-date.
jvoisin@kaa 23:15 /opt/aircrack-ng make clean
rm -rf cov-int
make -C src clean
make[Entering directory '/opt/aircrack-ng/src'
make -C osdep clean
make[2](1]:): Entering directory '/opt/aircrack-ng/src/osdep'
make -C radiotap clean
make[Entering directory '/opt/aircrack-ng/src/osdep/radiotap'
rm -f *.o
make[3](3]:): Leaving directory '/opt/aircrack-ng/src/osdep/radiotap'
rm -f libosdep.a  *.o .os.*
make[Leaving directory '/opt/aircrack-ng/src/osdep'
rm -f aireplay-ng airodump-ng airserv-ng airtun-ng airbase-ng aircrack-ng airdecap-ng packetforge-ng ivstools kstats makeivs-ng airdecloak-ng wpaclean besside-ng-crawler aircrack-ng-opt-prof_gen aircrack-ng-opt aircrack-ng-opt-prof prof/* airolib-ng *.o wesside-ng tkiptun-ng easside-ng buddy-ng besside-ng a.out liblinecount.a
make[1](2]:): Leaving directory '/opt/aircrack-ng/src'
make -C test/cryptounittest clean
make[Entering directory '/opt/aircrack-ng/test/cryptounittest'
rm -f test-encrypt-wep test-calc-pmk test-calc-mic test-calc-ptk test-encrypt-ccmp test-decrypt-ccmp test-calc-4pmk
rm -f vgcore*
make[1](1]:): Leaving directory '/opt/aircrack-ng/test/cryptounittest'
make -C test clean
make[Entering directory '/opt/aircrack-ng/test'
rm -f test-hex_string_to_array.o test-hex_string_to_array
make[1](1]:): Leaving directory '/opt/aircrack-ng/test'
jvoisin@kaa 23:15 /opt/aircrack-ng make
make -C src all
make[Entering directory '/opt/aircrack-ng/src'
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o aircrack-ng.o aircrack-ng.c
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o crypto.o crypto.c
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o common.o common.c
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o uniqueiv.o uniqueiv.c
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude   -c -o aircrack-ptw-lib.o aircrack-ptw-lib.c
g++ -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude -masm=intel -fdata-sections -ffunction-sections   -c -o linecount.o linecount.cpp
ar cru liblinecount.a linecount.o
ar: `u' modifier ignored since `D' is the default (see `U')
ranlib liblinecount.a
touch liblinecount.a
gcc -fstack-protector-strong -Wformat -Werror=format-security -Wstrict-overflow -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=0  -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude  aircrack-ng.o crypto.o common.o uniqueiv.o aircrack-ptw-lib.o liblinecount.a sha1-sse2.S -o aircrack-ng -lpthread -lssl -lcrypto   -lstdc++
aircrack-ng.o: In function `crack_wpa_thread':
aircrack-ng.c:(.text+0xb66d): undefined reference to `wpa_receive_passphrase'
aircrack-ng.o: In function `do_wpa_crack':
aircrack-ng.c:(.text+0xd3e8): undefined reference to `wpa_send_passphrase'
aircrack-ng.c:(.text+0xd46e): undefined reference to `wpa_send_passphrase'
collect2: error: ld returned 1 exit status
Makefile:142: recipe for target 'aircrack-ng' failed
make[1](1]:): *** [Error 1
make[1](aircrack-ng]): Leaving directory '/opt/aircrack-ng/src'
Makefile:25: recipe for target 'all' failed
make: *** [all] Error 2
zsh: exit 2     make
jvoisin@kaa 23:15 /opt/aircrack-ng
```

**Comment by aircrack-ng** [comment]
**Comment by Zero Chaos on 11 Dec 2015 22:18 UTC**

same line that fails but from a working build

x86_64-pc-linux-gnu-gcc -Os -mtune=nocona -pipe -frecord-gcc-switches -march=native -D_FILE_OFFSET_BITS=64 -D_REVISION=2774  -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -D_FILE_OFFSET_BITS=64 -D_REVISION=2774  -DHAVE_SQLITE  -DHAVE_PCRE -DCONFIG_LIBNL30 -DCONFIG_LIBNL -I/usr/include/libnl3  -fstack-protector-strong -Wno-unused-but-set-variable -Wno-array-bounds -Iinclude -Wl,-O1 -Wl,--as-needed -Wl,--hash-style=gnu aircrack-ng.o crypto.o common.o uniqueiv.o aircrack-ptw-lib.o liblinecount.a sha1-sse2.S -o aircrack-ng -lpthread -lssl -lcrypto -Wl,-O1 -Wl,--as-needed -Wl,--hash-style=gnu -L/usr/local/lib -lsqlite3 -lstdc++

**Comment by aircrack-ng** [comment]
**Comment by misterx on 11 Dec 2015 22:18 UTC**

Only happens when disabling/changing optimization. Fixed in aircrack-ng/aircrack-ng@344bc7208f709c0fdad885b300ab1ccd82596900.
